app/helpers/validation_helper.php

```diff
@@ -0,0 +1,21 @@
+<?php
+function validate_between($str, $min, $max)
+{
+    $length = mb_strlen($str); // count characters based on encoding
+    return $length <= $max && $length >= $min;
+}
+
+function validate_username($str)
+{
+    return preg_match('/^[[:alnum:]_]*$/', $str);
+}
+
+function validate_unique_name($username)
+{
+    return !User::getByUsername($username);
```

Comment:
You can catch the duplicate entry exception and to set the $user validation errors to true so you can display the error message in the view po :bow:
```php 
$user->validation_errors['username']['unique'] = true;
```